Instrukcije_d001/src/domain/World.c

```diff
@@ -7,11 +7,76 @@
 World world_new() { // ker imamo fixno dolzino ne rabimo parametrov
     World this;
 
-    this.height = 3;
-    this.width = 3;
+    // Vedno pri prirejanju elementov struct dodaj type cast
+    this = (World){.height = 3, .width = 3, .active = 0, .frames = 0, .memoTabela = {{0,0,0}, {0,0,0}, {0,0,0}}};
+
     for (int i = 0; i < WORLD_ST_ZETONOV; i++) {
         this.zetoni[i].aktiven = 0;
     }
 
     return this;
+}
+
+// (return value) 0 - no winner detected, 1 - winner detected, 2 - it's a draw
+int dinamicnoPoisciZmagovalca(int x, int y, int enaALIdve, int memoTabela[SIRINA_BOARDA][SIRINA_BOARDA], World* world) {
```

Comment:
Če imaš file World.c potem vse funkcije poimenuj z prependanim world_ da tako poveč da ta funkcija pripada world strukturi, verjemi mi če imaš 30 fileov v domenski logiki in v vsakem fileu vsaj 5 funkcij potem si boš samemu sebi zelo hvaležen.

Reply:
Pa še to strukturo za katero primerjaš je zmeraj na prvem mestu
zelo podobno kot to dela python classes...
```
class Dog:
    species = "Canis familiaris"

    def __init__(self, name, age):
        self.name = name
        self.age = age

    # Instance method
    def description(self):
        return f"{self.name} is {self.age} years old"

    # Another instance method
    def speak(self, sound):
        return f"{self.name} says {sound}"
```

A vidiš pri vsaki metodi je self na prvem mestu in naj bo to standardizirano po celotnem projektu.